tests/integration/test_fim/test_files/test_wildcards_complex/test_wildcards_complex.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import os
+import sys
+import pytest
+from wazuh_testing import global_parameters
+from wazuh_testing.fim import LOG_FILE_PATH, regular_file_cud, generate_params
+from wazuh_testing.tools import PREFIX
+from wazuh_testing.tools.configuration import load_wazuh_configurations, check_apply_test
+from wazuh_testing.tools.monitoring import FileMonitor
+
+# Marks
+
+pytestmark = pytest.mark.tier(level=1)
+
+# Variables
+
+test_folder = os.path.join(PREFIX, 'test_folder')
+test_directories = [test_folder]
+
+matched_dirs = [os.path.join('stardir', 'sub_test'), os.path.join('multiple_wildcards', 'sub_test'),
+                os.path.join('directory_test', 'test_subdir1'), os.path.join('test_all', 'testdir'),
+                os.path.join('test_all', 'testdir', 'all')]
+
+no_match_dirs = ['random_directory']
+
+wildcards = ','.join([os.path.join(test_folder, 'star*', 'sub*'), os.path.join(test_folder, 'mul*', '*test'),
+                      os.path.join(test_folder, '*test*', '*dir?'), os.path.join(test_folder, 'test_all', '*'),
+                      os.path.join(test_folder, 'test_all', '*', '*')])
+
+test_subdirectories = matched_dirs + no_match_dirs
+
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_conf_wildcards.yml')
+
+# Configurations
+
+conf_params = {'TEST_WILDCARDS': wildcards}
+parameters, metadata = generate_params(extra_params=conf_params)
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+
+
+def extra_configuration_before_yield():
+    for dir in test_subdirectories:
+        split_path = os.path.split(dir)
+        parent_folder = os.path.join(test_folder, split_path[0])
+        if not os.path.exists(parent_folder):
+            os.mkdir(os.path.join(parent_folder))
+        if not os.path.exists(os.path.join(parent_folder, split_path[1])):
+            os.mkdir(os.path.join(parent_folder, split_path[1]))
```

Comment:
Not sure about this but.. wouldn't it be more clear if we just create dir instead of `os.path.join(parent_folder, split_path[1])`? 

Also, I see that we need to create that kind of folder structure in various tests, why not create a function in the file module in our dependencies tools? I would create a function there to build a folder structure and use it in all tests. Also, I would try to make this function recursive because we may want to add more directories in the future and this only would work for directories with depth=2.